tracing-core/src/subscriber.rs

```diff
@@ -283,12 +283,26 @@ pub trait Subscriber: 'static {
     /// what that means for the specified pointer.
     ///
     /// [span ID]: ../span/struct.Id.html
-    /// [`drop_span`]: trait.Subscriber.html#method.drop_span
+    /// [`try_close`]: trait.Subscriber.html#method.drop_span
     fn clone_span(&self, id: &span::Id) -> span::Id {
         id.clone()
     }
 
-    /// Notifies the subscriber that a [span ID] has been dropped.
+    /// **This method is soft-deprecated.**
+    ///
+    /// Although using it won’t cause compilation warning, new code should
+    /// call or implement [`try_close`] instead.
+    ///
+    /// The default implementation of this function simply calls `try_close` and
+    /// ignores the return value.
+    ///
+    /// [`try_close`]: trait.Subscriber.html#method.try_close
+    fn drop_span(&self, id: span::Id) {
```

Comment:
Maybe I'm missing something here.. If the implementation is updated to call `try_close` instead of `drop_span`, then existing `Subscriber` implementations that only implement `drop_span` would never see that function be called, since `try_close` does not forward to `drop_span`. That seems like a breaking change, again assuming that we'll now start calling `try_close` instead of `drop_span` in `tracing`.

Reply:
Ah, yeah, that might be an issue. I think `try_close` should default to calling `drop_span` instead?

Reply:
Possibly, but that is an issue too, no -- what do you return from it?

Reply:
@jonhoo it should return `false`. If the subscriber doesn't have a way of indicating that the span should be closed, then close notifications will not occur. In a world where `try_close` defaults to calling `drop_span`, we should probably deprecate manual impls of drop_span for this reason.